.github/workflows/build_cmake.yml

```diff
@@ -0,0 +1,62 @@
+name: Linux/MacOS Build
+
+on: [push, pull_request]
+
+#env:
+
+jobs:
+  build:
+    runs-on: [self-hosted, Linux]
+    strategy:
+      fail-fast: false
+      matrix:
+        build_type : [ Debug ]
```

Comment:
We don't want to test Release at this time? Or Profiling etc...?

This could come in an additional PR to move forward.

Reply:
It was taking too long on my machine, so I removed everything but one build. Once we have more runners we can enable a wider range of tests.